kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemMetadataProvider.java

```diff
@@ -534,6 +535,16 @@ static void writeDescriptor(
         throw new DatasetIOException("Cannot close", e);
       }
     }
+
+    //rename the file into its final location so that changes appear atomic
+      try {
+          fs.rename(tempDescriptorPath, descriptorPath);
```

Comment:
Using FileContext the rename w/ overwrite is atomic for HDFS.  It is a bit confusing to trace through the code as there is a call to AbstractFileSystem.rename() which has a non-atomic default implementation.  For HDFS this is overridden in org.apache.hadoop.fs.Hdfs, which contains an atomic implementation.